table/block_prefix_index.cc

```diff
@@ -189,6 +189,12 @@ Status BlockPrefixIndex::Create(const SliceTransform* internal_prefix_extractor,
           "Corrupted prefix meta block: unable to read from it.");
       break;
     }
+    // TODO DURING REVIEW: without this, ingested sstables without any keys
+    // other than range deletions hit an assertion in BlockPrefixIndex::Builder.
+    // I need to figure out why this is.
+    if (num_blocks == 0) {
```

Comment:
Is a prefix meta block created even if there are no keys, only range deletions? That doesn't seem right.

Reply:
It looks like it is. I was hoping that this was a result of https://github.com/facebook/rocksdb/pull/3906, but the block is definitely create, even when there are no keys. I don't actually see any logic that would prevent that. The meta block written in [`HashIndexBuilder::Finish`](https://github.com/facebook/rocksdb/blob/aa53579d6cc5a53c1493bbda4f21c0e9831b377b/table/index_builder.h#L234), and this is called from [`BlockBasedTableBuilder::Finish`](https://github.com/facebook/rocksdb/blob/aa53579d6cc5a53c1493bbda4f21c0e9831b377b/table/block_based_table_builder.cc#L687) regardless of whether `empty_data_block` is true or not. Is it possible that this code path has just never been hit before, even though SSTs with only range deletions were already supported?

Reply:
> Is it possible that this code path has just never been hit before, even though SSTs with only range deletions were already supported?

Actully, I think this might be is exactly what's going on. See https://github.com/facebook/rocksdb/issues/3774, which is the same assertion I hit in my test without this patch.